docs/source/gcp/examples/appengine-firewall-ingress-rule.rst

```diff
@@ -0,0 +1,60 @@
+App Engine - Check if a Firewall Rule is in Place
+==================================================
+Custodian can check and notify if App Engine firewall ingress rules have been mis-configured. Note that the ``notify`` action requires a Pub/Sub topic to be configured.
+
+In the example below, the policy checks that there is only one rule allowing all connections.
+
+.. code-block:: yaml
+
+    policies:
+      - name: gcp-app-engine-firewall-ingress-rule-notify-if-default-unrestricted-access
+        resource: gcp.app-engine-firewall-ingress-rule
+        filters:
+          - and:
+            - type: value
+              value_type: resource_count
+              op: eq
+              value: 1
+            - type: value
+              key: sourceRange
+              value: '*'
+            - type: value
+              key: action
+              value: ALLOW
+        actions:
+          - type: notify
+             to:
+               - email@address
+             subject: App Engine has default unrestricted access
+             format: txt
+             transport:
+               type: pubsub
+               topic: projects/my-gcp-project/topics/my-topic
+
+
+In this variant, the policy checks if a custom DENY rule is in place and notify if absent.
+
+.. code-block:: yaml
+
+    policies:
+        - name: appengine-firewall-rules
+          description: |
+            Check absence of a required firewall rule
+          resource: gcp.app-engine-firewall-ingress-rule
+          filters:
+          - type: value
+            key: sourceRange
+            value: 192.168.2.0/24
+            op: absent
```

Comment:
value_type: cidr  is also useful for ip range checks, and allow container comparison operators (in/not-in) etc.

Reply:
Thank you for pointing that out. As `sourceRange` is not strictly CIDR (can have a * or an IP address without mentioning network prefix length), I have come up with a more complex example that takes this fact into account and uses cidr_size.